src/blocking/spi.rs

```diff
@@ -104,3 +104,31 @@ pub mod write_iter {
         }
     }
 }
+
+
+/// Actions for transactional SPI trait
+/// 
+/// This allows composition of SPI operations into a single bus transaction
+#[cfg(feature = "unproven")]
+#[derive(Debug, PartialEq)]
+pub enum Actions<'a> {
+    /// Read data into the provided buffer, write data undefined
+    Read(&'a mut [u8]),
+    /// Write data from the provided buffer, discarding read data
+    Write(&'a [u8]),
+    /// Transfer data from the provided buffer, overwriting the buffer with read data
+    Transfer(&'a mut [u8]),
+}
```

Comment:
What about an `Ignore(u8, usize)`, which clocks out a number of identical bytes and discards the read data?

I've worked with FPGAs (specifically: the ICE40 series) which require some dummy cycles at certain points in their initialization sequence. While this _could_ be done with `Write`, that would require allocating an output buffer when not truly necessary.

Reply:
I think this is probably out of scope for this crate, requiring all hal implementations to add an `Ignore` method (which is not something actually supported by hardware afaik) seems like more trouble than allocating a buffer if you need this functionality.

Reply:
That's a fair point

Reply:
relates to https://github.com/rust-embedded/embedded-hal/issues/52